### PR TITLE
Decode entities in site title

### LINKED ISSUE #33321: Site title renders escaped characters in multiple locations
## Description
Sites with a site title that contains characters that are beeing escaped shows the encoded enties multiple locations in the backend.

- `readOnlyTitle` in the `Site Title` block.
- In the post publish panel
- In the site editor navigation sidebar.

## Step-by-step reproduction instructions
1. Update the site title to ex. `Compony Name & "Son"`
2. Create a new post and click publish
3. In the post publish sitebar the site title is showing `Componany Name &amp; &quot;Son&quot;`

## Expected behaviour
Entities are decoded.

## Actual behaviour
Entities are shown.

## Screenshots or screen recording (optional)
image

image

image

## WordPress information
- WordPress version: 5.8 RC2
- Gutenberg version: 11.0 / and deactivated
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes. Twenty Twenty-One / tt1

## Device information
- Device: Desktop
- Operating system: macOS Big Sur
- Browser: Chrome 91

### PR DESCRIPTION
**Pull request by walbo**
## Description
Fixes #33321

## How has this been tested?
Locally. Added a site title that have characters that are going to be encoded and confirmed is shows correctly in the locations described in #33321.

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/site-title/edit/index.js [hunk 1]
```diff
@@ -34,7 +35,7 @@ export default function SiteTitleEdit( {
 		const siteData = getEntityRecord( 'root', '__unstableBase' );
 		return {
 			canUserEdit: canUser( 'update', 'settings' ),
-			readOnlyTitle: siteData?.name,
+			readOnlyTitle: siteData?.name && decodeEntities( siteData.name ),
```
Comment (Mamaduka) [inline packages/block-library/src/site-title/edit/index.js]:
```suggestion
			readOnlyTitle: decodeEntities( siteData?.name ),
```

I think we can just pass data here no need for an extra check. The `decodeEntities` will return the passed value if it's not a string.

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js [hunk 2]
```diff
@@ -48,7 +49,7 @@ const NavigationPanel = ( { isOpen } ) => {
 			editedPostId: getEditedPostId(),
 			editedPostType: getEditedPostType(),
 			activeMenu: getNavigationPanelActiveMenu(),
-			siteTitle: siteData.name,
+			siteTitle: siteData.name && decodeEntities( siteData.name ),
```
Comment (Mamaduka) [inline packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js]: Maybe we should decode the site title before rendering. I think this a pattern generally used in the codebase.

https://github.com/WordPress/gutenberg/blob/28db73d54bf30fc73886e267174aee6343a01e8c/packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js#L98

File: packages/editor/src/components/post-publish-panel/prepublish.js [hunk 3]
```diff
@@ -53,7 +54,7 @@ function PostPublishPanelPrepublish( { children } ) {
 				undefined,
 			] ),
 			siteIconUrl: siteData.site_icon_url,
-			siteTitle: siteData.name,
+			siteTitle: siteData.name && decodeEntities( siteData.name ),
```
Comment (Mamaduka) [inline packages/editor/src/components/post-publish-panel/prepublish.js]: Same here. Let's decode before rendering.

https://github.com/WordPress/gutenberg/blob/28db73d54bf30fc73886e267174aee6343a01e8c/packages/editor/src/components/post-publish-panel/prepublish.js#L108

**Review by Mamaduka — CHANGES_REQUESTED** [review]
The fix works as expected. I just request few minor changes hope you don't mind. 🙇

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js (hunk 2 above)
Comment (walbo) [reply packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js]: Sure. Lets use the same pattern that is used elsewhere. I will update.

**Comment by Mamaduka** [comment]
@walbo, sorry for the confusion. I didn't recommend decoding `readOnlyTitle` on render because it's used in a few places and decoding once, in this case, makes more sense.

**Comment by walbo** [comment]
Ah. My mistake. I will revert that one.

**Review by Mamaduka — APPROVED** [review]
This looks great, and thanks for the quick updates, @walbo.

It's always a pleasure reviewing your PRs.

**Pull request merged**